### PR TITLE
DALA-4311 Fast Frontend Development for all (even on windows)

### PR DESCRIPTION
**Pull request by MarcTM01**
…r docker stack

# Pull Request \<Title>
`<Description here>`
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
  - [ ] If this PR includes work on the frontend, at least one `@ts-nocheck` is removed. Additionally, there should not be any `@ts-nocheck` in files modified by this PR. If no `@ts-nocheck` are left: Celebrate :tada: :confetti_ball: type-safety and remove this entry. 
- [x] The PR actually implements what is described in the JIRA-Issue
- [ ] At least one test exists testing the new feature
  - [ ] If you have created new test files, make sure that they are included in a test container and actually run in the CI
- [x] At least 3 consecutive CI runs are successfully executed to ensure that there are no flaky tests.
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [ ] If there was a database entity class added, there must also be a migration script for creating the corresponding database if flyway is already used by the service
- [ ] IF there are changes done to the framework data models or to a database entity class, the following steps were completed in order
  - [ ] A fresh clone of dataland.com is generated (see Wiki page on "OTC" for details)
  - [ ] The feature branch is deployed to clone with `Reset non-user related Docker Volumes & Re-populate` turned off
  - [ ] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on the clone server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team
  - [ ] The feature branch is deployed to dev1 with `Reset non-user related Docker Volumes & Re-populate` turned on, and it's verified that the CD run is green  
- [x] ELSE, the new version is deployed to the dev server "dev1" using this branch
  - [x] Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [x] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [x] It's verified that the CD run is green
  - [x] The new feature is manually used/tested/observed on dev server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team

### CODE REVIEW

File: startDevelopmentStack.sh [hunk 1]
```diff
@@ -1,5 +1,35 @@
...
+  case "$opt" in
+    l)
+      echo "Launching in local frontend mode."
+      local_frontend=true
+      export FRONTEND_LOCATION_CONFIG="Localhost"
+      ;;
+
+    h)
+      echo "Usage: $(basename $0) [-l] [-h]"
+      echo "  -l: Run in local frontend mode. Do not launch a frontend docker container. Instead, redirect frontend traffic to localhost."
```
Comment (stewal19) [inline startDevelopmentStack.sh:15]: Maybe add the info that you have to start the frontend locally with npm run dev, so that is somewhere documented.

Comment (MarcTM01) [reply startDevelopmentStack.sh:15]: A great idea. I even went a little bit further and added an extra error page to the proxy for exactly this use-case. That page gets displayed when the frontend is not reachable when you are running in local frontend mode. Feel free to check it out

Comment (stewal19) [reply startDevelopmentStack.sh:15]: I like the idea! Will check it out

File: dataland-inbound-proxy/config/utils/qaErrors.conf [hunk 2]
```diff
@@ -4,5 +4,6 @@ location = /qa/json_errors.json {
   ssi on;
   ssi_types application/json;
   internal;
-  root /var/www/html;
+  root /var/www/html/error_documents;
+  try_files /json_errors.json =500;
```
Comment (stewal19) [inline dataland-inbound-proxy/config/utils/qaErrors.conf:8]: Was macht das genau?

Comment (MarcTM01) [reply dataland-inbound-proxy/config/utils/qaErrors.conf:8]: Die error pages haben nicht funktioniert, weil die versuchen auf /api/json_errors.json in dem Verzeichniss zuzugreifen. Mit try files leite ich das auf /json_errors.json um. Damit klappt das dann wieder

Comment (stewal19) [reply dataland-inbound-proxy/config/utils/qaErrors.conf:8]: Und was macht der =500 Teil?

Comment (MarcTM01) [reply dataland-inbound-proxy/config/utils/qaErrors.conf:8]: Wenn die Datei `json_errors.json` aus irgendwelchen Gründen nicht geunden wird, wird ein 500 - Internal Server Errror zurückgegeben (siehe http://nginx.org/en/docs/http/ngx_http_core_module.html#try_files). Es ist zwingend notwendig mindestens zwei Möglichkeiten angeben, deshalb habe ich das `=500` angefügt. Probiere es gerne bei dir lokal aus: Starte den stack, kille das backend, und gehe dann auf https://local-dev.dataland.com/api/xasd. Dann solltest du die json-formatierte Fehlermeldung sehen.

**Pull request merged**